### DISCUSSION TITLE
Add folder navigation

### DISCUSSION THREAD

**Discussion by mlepannetier** [navidrome/navidrome#2414] [Ideas]
Hi,

Please add folder navigation, now i use airsonic and this function is essential

Thank's

**Comment by dweymouth** [comment]
This is currently outside Navidrome's goals of being a tags-oriented music server. If you're looking to migrate away from Airsonic you could give Gonic a try, which does implement the browse-by-folders API.

**Comment by Glandos** [comment]
https://www.navidrome.org/docs/faq/#-can-you-add-a-browsing-by-folder-optionmode-to-navidrome is quite inconsistent. It says:
> While it is technically possible to add a browsing by folder option, doing so would require significant changes to Navidrome’s internal structures across most of its components.

Yet, recent release https://github.com/navidrome/navidrome/releases/tag/v0.58.0 mentionned:
> Navidrome v0.58.0 introduces [Multi-library Support](https://www.navidrome.org/docs/usage/multi-library/), a major architectural enhancement […].

So, yes, restructuring is not done on purpose.

However, this introduce a vendor lock-in, that is unbearable. A lot of people organize their collections by folder, for at least 2 reasons:
1. They don't have Navidrome or similar before discovering it.
2. They don't always have Navidrom or similar. I personally uses Gonic only when I'm outside home. Otherwise I have a NFS share with a local player, which is much more performant than any web based interface.

I like tags and classification, but folders is one of them, and it's the only one that is supported on every OSes with simple tools.

**Reply by benjamindaines** [reply]
As the documentation says, Navidrome is built around tags.  If you library is untagged, you're going to run into this issue with a lot of different servers or players.  The best option would be to go through and tag your library.  I doubt that folder browsing will end up getting added, as the documentation explicitly says that it is against the entire point / design of Navidrome.  

> If browsing by folder is an essential feature for you, there are [alternative music servers](https://github.com/basings/selfhosted-music-overview) that offer this functionality. We encourage you to explore these options if folder browsing is a priority.

It may not be the answer you want, but, again, I would say that it's very worth having a properly tagged library.  It was a pain to do when I stopped using Roon (Roon maintains it's own database separate from file tags, so it encouraged me to be lazy and not tag my music properly), but now I just keep up with it as I add new music to my library.
